### PR TITLE
Implemented lookup scan in storage layer

### PR DESCRIPTION
**Pull request by bright-starry-sky**
Because #129 branch error. re-submit the PR instead of #129.

### CODE REVIEW

**Comment by bright-starry-sky** [comment]
Removed timezone of DateTime since common layer changed.

File: src/storage/exec/IndexFilterNode.h [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+                    data_.emplace_back(k.first, k.second);
+                }
+            }
+        }
+        return kvstore::ResultCode::SUCCEEDED;
+    }
+
+    const std::vector<kvstore::KV>& getData() const {
+        return std::move(data_);
+    }
```
[the remark this answers is not in the document]
Comment (bright-starry-sky) [reply src/storage/exec/IndexFilterNode.h]: My fault . A very comprehensive example. Thanks.

**Comment by bright-starry-sky** [comment]
1, Addressed dutor's comments; 2, changed the index prop to big endian for date and datetime.

**Review by dutor — APPROVED** [review]
LGTM.

**Review by critical27 — APPROVED** [review]
LGTM

File: src/storage/exec/IndexOutputNode.h [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+
+    kvstore::ResultCode edgeRowsFromIndex(const std::vector<kvstore::KV>& data) {
+        auto returnCols = result_->colNames;
+        for (const auto& val : data) {
+            Row row;
+            auto src = IndexKeyUtils::getIndexSrcId(planContext_->vIdLen_, val.first);
+            auto rank = IndexKeyUtils::getIndexRank(planContext_->vIdLen_, val.first);
+            auto dst = IndexKeyUtils::getIndexDstId(planContext_->vIdLen_, val.first);
+
+            row.emplace_back(Value(std::move(src)));
```
Comment (panda-sheep) [inline src/storage/exec/IndexOutputNode.h:247]: Is it ` row.values.emplace_back(std::move(src))` ？

File: src/storage/exec/IndexOutputNode.h [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+            row.emplace_back(Value(std::move(rank)));
+            row.emplace_back(Value(std::move(dst)));
```
Comment (panda-sheep) [inline src/storage/exec/IndexOutputNode.h:249]: ditto

By the way, the same everywhere else.

File: src/storage/exec/IndexOutputNode.h (hunk 2 above)
Comment (dutor) [reply src/storage/exec/IndexOutputNode.h:247]: `Row`, a.k.a `List` has a delegate on this method.

Comment (panda-sheep) [reply src/storage/exec/IndexOutputNode.h:247]: 👍，this method is concise

**Pull request merged**